oa/proxy/proxy_oa.py

```diff
@@ -0,0 +1,291 @@
+            iana_config  = json.loads(open(iana_conf_file).read())
+            proxy_iana = IanaTransform(iana_config["IANA"]) 
+
+        for conn in self._proxy_scores:
+            conn_hash = conn[self._conf["proxy_score_fields"]["hash"]]
+            if conn_hash not in hash_list:
+                hash_list.append(conn_hash)
+                clientip = conn[self._conf["proxy_score_fields"]["clientip"]]
+                fulluri = conn[self._conf["proxy_score_fields"]["fulluri"]]
+                date=conn[self._conf["proxy_score_fields"]["p_date"]].split('/')
+                if len(date) == 3:
+                    year=date[2]
+                    month=date[0].zfill(2)
+                    day=date[1].zfill(2)                
+                    hh=(conn[self._conf["proxy_score_fields"]["p_time"]].split(":"))[0]
+                    # print hh
+                    self._get_proxy_details(fulluri,clientip,conn_hash,year,month,day,hh,proxy_iana)
+
+
+    def _get_proxy_details(self,fulluri,clientip,conn_hash,year,month,day,hh,proxy_iana):
+                    
+        limit = 250
+        output_delimiter = '\\t' 
+        edge_file ="{0}/edge-{1}_{2}.csv".format(self._data_path,clientip,conn_hash)
+        edge_tmp  ="{0}/edge-{1}_{2}.tmp".format(self._data_path,clientip,conn_hash)
+
+        if not os.path.isfile(edge_file):
+            proxy_qry = ("SELECT p_date, p_time, clientip, host, webcat, respcode, reqmethod, useragent, resconttype, \
+                referer, uriport, serverip, scbytes, csbytes, fulluri FROM {0}.{1} WHERE y={2} AND m={3} AND d={4} AND \
+                h={5} AND fulluri =\'{6}\' AND clientip = \'{7}\' LIMIT {8};").format(self._db,self._table_name, year,month,day,hh,fulluri,clientip,limit)
+
+            # execute query
+            self._engine.query(proxy_qry,edge_tmp,output_delimiter) 
+            # add IANA to results.
+            self._logger.info("Adding IANA translation to details results")
+            with open(edge_tmp) as proxy_details_csv:
+                rows = csv.reader(proxy_details_csv, delimiter='\t', quotechar='|')
+                next(proxy_details_csv)
+                update_rows = [[conn[0]] + [conn[1]] + [conn[2]] + [conn[3]] + [conn[4]] + [proxy_iana.get_name(conn[5],"proxy_http_rcode")  if proxy_iana else conn[5]] + [conn[6]] + [conn[7]] + [conn[8]] + [conn[9]] + [conn[10]] + [conn[11]] + [conn[12]] + [conn[13]] + [conn[14]] for conn in rows]
+                update_rows = filter(None, update_rows)
+                header = ["p_date","p_time","clientip","host","webcat","respcode","reqmethod","useragent","resconttype","referer","uriport","serverip","scbytes","csbytes","fulluri"]
+                update_rows.insert(0,header)
+ 
+
+            # create edge file.
+            self._logger.info("Creating edge file:{0}".format(edge_file))
+            with open(edge_file,'wb') as proxy_details_edge:
+                writer = csv.writer(proxy_details_edge, quoting=csv.QUOTE_ALL)
```

Comment:
csv.writer should not be quoting any field. D3's csv parser is unable to handle quotation, any delimiter inside the quotation will end up as a new column on the UI :(